openstudiocore/src/model/MaterialPropertyGlazingSpectralData.hpp

```diff
@@ -0,0 +1,114 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_MATERIALPROPERTYGLAZINGSPECTRALDATA_HPP
+#define MODEL_MATERIALPROPERTYGLAZINGSPECTRALDATA_HPP
+
+#include <model/ModelAPI.hpp>
+#include "ResourceObject.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  class MaterialPropertyGlazingSpectralData_Impl;
+
+} // detail
+
+class MODEL_API SpectralDataField
```

Comment:
@jmarrec this was the pattern i was suggesting for extensible fields (e.g. in custom meter)